### PR TITLE
Show most frequently used blocks next to inserter

### PR DESCRIPTION
**Pull request by notnownikki**
## Description
Instead of hardcoded paragraph and image blocks, this shows
the three most frequently used blocks, based on stored blockUsage

The algorithm for working out the blocks to display is just "most frequently used." I think that's fine for now, but we probably want to think about future refinements that "age" the usage too, in the future.

## How to test
Hover over the inserter icon in the main editing section, and your three most used blocks should appear next to it. Click on them to insert them.

Clear your local storage, and paragraph and image should be displayed to start with.

### CODE REVIEW

File: editor/modes/visual-editor/test/inserter.js [hunk 1]
```diff
@@ -26,11 +26,17 @@ describe( 'VisualEditorInserter', () => {
 		expect( wrapper.state( 'isShowingControls' ) ).toBe( false );
 	} );
 
-	it( 'should insert paragraph block', () => {
+/*	it( 'should insert paragraph block', () => {
 		const onInsertBlock = jest.fn();
 		const wrapper = shallow(
 			<VisualEditorInserter onInsertBlock={ onInsertBlock } />
```
Comment (youknowriad) [inline editor/modes/visual-editor/test/inserter.js]: Actually we do not test the "connected" component but just the presentational one, so you just need to provide "mostFrequentlyUsedBlocks" as a prop.

**Review by youknowriad — COMMENTED** [review]
This is looking very good for me.

> I think the getMostFrequentlyUsedBlocks method needs to be memoized

Sounds like a good idea 👍

**Comment by notnownikki** [comment]
Memoize added, and tests fixed up for the inserter.

File: editor/selectors.js [hunk 2]
```diff
@@ -879,13 +880,15 @@ export function getRecentlyUsedBlocks( state ) {
...
 /**
  * Resolves the block usage stats into a list of the most frequently used blocks.
+ * Memoized so we're not generating block lists every time we render the list
+ * in the inserter.
  *
  * @param {Object} state Global application state
  * @return {Array}       List of block type settings
  */
-export function getMostFrequentlyUsedBlocks( state ) {
+export const getMostFrequentlyUsedBlocks = memoize( ( state ) => {
```
Comment (youknowriad) [inline editor/selectors.js]: Should we use `createSelector` helper instead to memoize the selector (we already use it elsewhere). https://github.com/aduth/rememo

Comment (notnownikki) [reply editor/selectors.js]: I was going to suggest something like this, I didn't see we already had something :) Will fix!

**Comment by youknowriad** [comment]
While I don't like hardcoding, I wonder if we should hard code the paragraph and image block as default if no usage yet (try clearing localstorage).

Also flagging this for design review, since it changes the quick blocks behavior.

**Comment by notnownikki** [comment]
Yeah I guess that's easy enough to do in the selector.

**Comment by karmatosed** [comment]
I really like this. For those looking into this, you can see here:

2017-10-09 at 21 08

**Review by karmatosed — APPROVED** [review]
⭐️

**Comment by aduth** [comment]
There's something disconcerting to me about the dynamic nature of this, that it's less predictable. Maybe I'm attached to the idea that there's some comfort in knowing I'd always find the same options in this control, an especially important comfort considering that the controls are hidden until revealed by hover.

Maybe I'm overthinking it 😄

**Comment by notnownikki** [comment]
Initially you'll notice changes, but my thinking was that over time, it'll stabilise, and at least he first two options will be pretty much static, but tailored for the user's usage.

**Pull request merged**